tracker/tracker.go

```diff
@@ -259,42 +228,56 @@ func (e *EventTracker) Start() error {
 		return err
 	}
 
-	go common.RetryForever(ctx, time.Second, func(context.Context) error {
-		// sync up all missed blocks on start if it is not already sync up
-		if err := e.syncOnStart(); err != nil {
-			e.config.Logger.Error("Syncing up on start failed.", "err", err)
+	e.config.Logger.Info("Starting event tracker",
+		"jsonRpcEndpoint", e.config.RPCEndpoint,
+		"numBlockConfirmations", e.config.NumBlockConfirmations,
+		"pollInterval", e.config.PollInterval,
+		"syncBatchSize", e.config.SyncBatchSize,
+		"numOfBlocksToReconcile", e.config.NumOfBlocksToReconcile,
+		"logFilter", e.config.LogFilter,
+	)
 
-			return handleError(err)
+	common.RetryForever(ctx, time.Second, func(context.Context) error {
+		// sync up all missed blocks on start if it is not already sync up
+		err := e.syncOnStart(ctx)
+		if err != nil {
+			return handleError(err, "Syncing up on start failed.")
 		}
 
 		// start the polling of blocks
-		err := e.blockTracker.Track(ctx, func(block *ethgo.Block) error {
-			return e.trackBlock(block)
+		err = e.blockTracker.Track(ctx, func(block *ethgo.Block) error {
+			return e.trackBlock(ctx, block)
 		})
 
-		if common.IsContextDone(err) {
-			return nil
-		}
-
-		return handleError(err)
+		return handleError(err, "Tracking blocks failed.")
 	})
 
+	e.config.Logger.Info("Event tracker stoped", "lastProcessedBlock", e.blockContainer.LastProcessedBlock())
+
 	return nil
 }
 
 // trackBlock is a method in the EventTracker struct that is responsible for tracking blocks and processing their logs
 //
 // Inputs:
 // - block: An instance of the ethgo.Block struct representing a block to track.
+// - ctx: A context.Context instance to manage cancellation and timeouts.
 //
 // Returns:
 //   - nil if tracking block passes successfully.
 //   - An error if there is an error on tracking given block.
-func (e *EventTracker) trackBlock(block *ethgo.Block) error {
+func (e *EventTracker) trackBlock(ctx context.Context, block *ethgo.Block) error {
 	if !e.blockContainer.IsOutOfSync(block) {
 		e.blockContainer.AcquireWriteLock()
 		defer e.blockContainer.ReleaseWriteLock()
 
+		if latestBlock := e.blockContainer.LastProcessedBlockLocked(); block.Number <= latestBlock {
```

Comment:
Should we maybe move this logic to the `blockContainer`, since it already has `IsOutOfSync`, and what not. Seems like a better place for this check in my opinion.